src/segment.rs

```diff
@@ -918,7 +918,8 @@ mod test {
         let castagnoli = crc::Crc::<u32>::new(&crc::CRC_32_ISCSI);
 
         (0..1024).for_each(|_| {
-            OsRng.fill_bytes(&mut buffer);
+            let mut rng = rand::rng();
```

Comment:
Let's move `let mut rng = rand::rng();` outside of the `for_each` block. That way we can probably reuse it across all three invocations.